torch/distributed/_tensor/dispatch.py

```diff
@@ -300,7 +301,16 @@ def unwrap_to_op_info(
                 else:
                     mesh = arg.device_mesh
             elif isinstance(arg, torch.Tensor):
-                if arg.ndim == 0 or self._allow_implicit_replication:
+                if arg.numel() == 1 and arg.ndim == 1:
+                    warnings.warn(
+                        "Found a non-scalar tensor with numel=1 and ndim!=0, "
+                        "we are implicitly creating a replicated DTensor for it. "
+                        "However, please consider changing it to a scalar tensor "
+                        "or explicitly create a DTensor under distributed enviroment."
+                    )
+
+                # if the arg.numel() == 1, arg.ndim could be 0 or 1.
+                if arg.numel() == 1 or self._allow_implicit_replication:
```

Comment:
Oh I think this should be `arg.ndim == 0`? and we can make this an `elif`

Reply:
We are turning the tensor with numel() = 1 to a replicated DTensor when arg.ndim <= 1. Hence, I didn't use an elif since we are appending the same args_schema for both cases. Therefore, I add an additional check before this to log the warnings when arg.ndim = 1.